src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -438,8 +438,42 @@ private function getOrderProducts(Order $order): OrderProductsForViewing
                 null;
             $product['product_quantity_refunded'] = $product['product_quantity_refunded'] ?: false;
 
+            $productType = !empty($product['pack_items']) ? OrderProductForViewing::TYPE_PACK :
+                OrderProductForViewing::TYPE_PRODUCT_WITHOUT_COMBINATIONS;
+
             $orderInvoice = new OrderInvoice($product['id_order_invoice']);
 
+            $packItems = [];
+            foreach ($product['pack_items'] as $pack_item) {
+                $packItemType = !empty($pack_item['pack_items']) ? OrderProductForViewing::TYPE_PACK :
+                    OrderProductForViewing::TYPE_PRODUCT_WITHOUT_COMBINATIONS;
+                $packItemImagePath = isset($pack_item['image_tag']) ?
+                    $this->imageTagSourceParser->parse($pack_item['image_tag']) :
+                    null;
+                $packItems[] = new OrderProductForViewing(
```

Comment:
What a constructor :sweat_smile:

Reply:
:smile:

Reply:
(nice-to-have) Could be improved by creating a dedicated constructor
```
$packItems[] = new OrderProductForViewing::createPackItem(...);
```

The constructor can either accepts all the arguments (at least we can remove the `null`) or a DTO